cmd/xl-v1-list-objects-heal.go

```diff
@@ -205,3 +206,237 @@ func (xl xlObjects) ListObjectsHeal(bucket, prefix, marker, delimiter string, ma
 	// Return error at the end.
 	return ListObjectsInfo{}, toObjectErr(err, bucket, prefix)
 }
+
+// ListUploadsHeal - lists ongoing multipart uploads that require
+// healing in one or more disks.
+func (xl xlObjects) ListUploadsHeal(bucket, prefix, marker, uploadIDMarker,
+	delimiter string, maxUploads int) (ListMultipartsInfo, error) {
+
+	if err := checkListObjsArgs(bucket, prefix, marker, delimiter, xl); err != nil {
+		return ListMultipartsInfo{}, err
+	}
+
+	switch {
+	// With max uploads of zero we have reached eof, return right here.
+	case maxUploads == 0:
+		return ListMultipartsInfo{}, nil
+	case maxUploads > maxUploadsList, maxUploads < 0:
+		return ListMultipartsInfo{}, InvalidMaxUpload{maxUploads}
+	}
+
```

Comment:
Just noticed that `ListUploadsHealHandler` is the only caller to `xl.ListUploadsHeal` - it appears that this checking of `maxUploads` is repeated in both functions. Also, why does the case of `maxUploads == 0` have to be handled at all? Shouldn't it be a user error and shouldn't the API spec also disallow this kind of useless exchange with the server?

For example, AWS list-multipart-uploads API does not accept zero for max-uploads - https://docs.aws.amazon.com/AmazonS3/latest/API/mpUploadListMPUpload.html